app/src/main/java/org/hackillinois/android/view/ScannerFragment.kt

```diff
@@ -1,215 +1,145 @@
-            null -> {
-                shouldParseScan = true
-                return
+        if (requestCode == PERMISSIONS_REQUEST_ACCESS_CAMERA) {
+            if (grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                codeScanner.startPreview()
             }
         }
+    }
 
+    override fun onResume() {
+        super.onResume()
+        codeScanner.startPreview()
+    }
+
+    override fun onPause() {
+        codeScanner.releaseResources()
+        super.onPause()
     }
 
-    /**
-     * Extracts the userId from the HackIllinois QR String URI.
-     * Given an example URI "hackillinois://user?userid=github0000001"
-     * it is enough to grab everything after the last (and only) equals sign.
-     *
-     * If the URI becomes more complex, we can use actual URI decoders.
-     * @return the userId from a QR String
-     */
     private fun getUserIdFromQrString(qrString: String): String {
         val splitOnEquals = qrString.split("=")
         return splitOnEquals.last()
     }
 
-    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<String>, grantResults: IntArray) {
-        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
-
-        if (requestCode == PERMISSIONS_REQUEST_ACCESS_CAMERA) {
-            if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
-                qrScanner.decodeContinuous(onQrCodeScan)
+    private fun displayScanResult(lastScanStatus: ScanStatus?) = lastScanStatus?.let {
+        val message = when (it.lastScanWasSuccessful) {
+            true -> {
+                staffOverrideSwitch.isChecked = false
+                "Success: ${lastScanStatus.userId}"
             }
+            false -> "User not registered, or already scanned in for event."
         }
-    }
-
-    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
-                              savedInstanceState: Bundle?): View? {
-        // Inflate the layout for this fragment
-        val view = inflater.inflate(org.hackillinois.android.R.layout.fragment_scanner, container, false)
 
-        view.eventListView.onItemSelectedListener = viewModel.onEventSelected
+        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
+    }
 
-        // Ensure the Camera permission is granted
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
-                && activity!!.checkSelfPermission(Manifest.permission.CAMERA)
-                != PackageManager.PERMISSION_GRANTED) {
-            requestPermissions(arrayOf(Manifest.permission.CAMERA),
-                    PERMISSIONS_REQUEST_ACCESS_CAMERA)
-        } else {
-            view.qrScanner.decodeContinuous(onQrCodeScan)
+    private fun updateOverrideSwitchVisibility(it: Boolean?) {
+        staffOverrideSwitch.visibility = when (it) {
+            true -> AdapterView.VISIBLE
+            false -> AdapterView.GONE
+            null -> AdapterView.GONE
```

Comment:
Group `false` and `null` together with `else`

Reply:
Good call